### PR TITLE
feat: second parameter in buy-coin

### PR DESCRIPTION
**Pull request by neutrinoks**

### CODE REVIEW

File: tests/assets/move-projects/car-wash-example/sources/CarWash.move [hunk 1]
```diff
@@ -46,18 +47,22 @@ module DeveloperBob::CarWash {
 
     /// Buys a washing coin for the car wash. Therfore, `COIN_PRICE` will be withdrawn from the user's account.
     // Note: It would be nice to have the ability to buy multiple coins at once, but it's not implemented for this example.
```
Comment (asmie) [inline tests/assets/move-projects/car-wash-example/sources/CarWash.move:49]: This comment should be updated.

File: tests/assets/move-projects/car-wash-example/sources/CarWash.move [hunk 2]
```diff
@@ -46,18 +47,22 @@ module DeveloperBob::CarWash {
...
 
+        // Verify, that enough coins are available.
+        let coins = borrow_global<Balance>(MODULE_OWNER).coins;
+        assert!(coins >= count, NOT_ENOUGH_COINS_AVAILABLE);
+
         // Transfer coin price from user to car-wash-and-module-owner
-        balance::transfer(user, MODULE_OWNER, COIN_PRICE);
+        balance::transfer(user, MODULE_OWNER, (count as u128)*COIN_PRICE);
 
         // After success, we deposit one more washing coin at the user's account.
```
Comment (asmie) [inline tests/assets/move-projects/car-wash-example/sources/CarWash.move:64]: This also needs update.

File: tests/assets/move-projects/car-wash-example/sources/CarWash.move [hunk 3]
```diff
@@ -68,26 +73,26 @@ module DeveloperBob::CarWash {
         assert!(exists<Balance>(user_addr), USER_DOES_NOT_EXIST);
 
         // Successful transfer of one coin will automatically start the washing process.
-        transfer_coin(user_addr, MODULE_OWNER);
+        transfer_coin(user_addr, MODULE_OWNER, 1);
     }
 
     /// Generic coin transfer function. Transfers always exactly one washing coin.
```
Comment (asmie) [inline tests/assets/move-projects/car-wash-example/sources/CarWash.move:79]: This comment also is outdated.

File: tests/assets/move-projects/car-wash-example/sources/CarWash.move [hunk 4]
```diff
@@ -68,26 +73,26 @@ module DeveloperBob::CarWash {
...
     /// Requires both accounts to exist! For module internal usage only!
-    fun transfer_coin(src: address, dst: address) acquires Balance {
+    fun transfer_coin(src: address, dst: address, count: u8) acquires Balance {
         // Check that source account has at least one washing coin.
```
Comment (asmie) [inline tests/assets/move-projects/car-wash-example/sources/CarWash.move:82]: And this.

File: tests/assets/move-projects/car-wash-example/sources/CarWash.move [hunk 5]
```diff
@@ -68,26 +73,26 @@ module DeveloperBob::CarWash {
...
         let coins_src = borrow_global<Balance>(src).coins;
-        assert!(coins_src > 0, NO_COINS_AVAILABLE);
+        assert!(coins_src > count, NO_COINS_AVAILABLE);
 
         // Check that destination has less than maximum number of coins.
         let coins_dst = borrow_global<Balance>(dst).coins;
-        assert!(coins_dst < 255, COIN_LIMIT_REACHED);
+        assert!(coins_dst + count < 255, COIN_LIMIT_REACHED);
 
         // Withdraw one washing coin.
```
Comment (asmie) [inline tests/assets/move-projects/car-wash-example/sources/CarWash.move:90]: Also :smile:

File: tests/assets/move-projects/car-wash-example/sources/CarWash.move [hunk 6]
```diff
@@ -68,26 +73,26 @@ module DeveloperBob::CarWash {
...
         let coins_ref = &mut borrow_global_mut<Balance>(src).coins;
-        *coins_ref = coins_src - 1;
+        *coins_ref = coins_src - count;
 
         // Deposit one washing coin.
```
Comment (asmie) [inline tests/assets/move-projects/car-wash-example/sources/CarWash.move:94]: :point_up:

File: doc/tutorial.md [hunk 7]
```diff
@@ -45,7 +48,7 @@ module DeveloperBob::CarWash {
 
     /// Buys a washing coin for the car wash. Therfore, `COIN_PRICE` will be withdrawn from the user's account.
     // Note: It would be nice to have the ability to buy multiple coins at once, but it's not implemented for this example.
```
Comment (asmie) [inline doc/tutorial.md:50]: This needs to be updated.

**Review by asmie — CHANGES_REQUESTED** [review]
Comments needs updating.

File: tests/assets/move-projects/car-wash-example/sources/CarWash.move (hunk 1 above)
Comment (neutrinoks) [reply tests/assets/move-projects/car-wash-example/sources/CarWash.move]: Done

File: tests/assets/move-projects/car-wash-example/sources/CarWash.move (hunk 2 above)
Comment (neutrinoks) [reply tests/assets/move-projects/car-wash-example/sources/CarWash.move:63]: Done

File: tests/assets/move-projects/car-wash-example/sources/CarWash.move (hunk 3 above)
Comment (neutrinoks) [reply tests/assets/move-projects/car-wash-example/sources/CarWash.move:78]: Why? :upside_down_face:

File: tests/assets/move-projects/car-wash-example/sources/CarWash.move (hunk 5 above)
Comment (neutrinoks) [reply tests/assets/move-projects/car-wash-example/sources/CarWash.move]: Done

File: doc/tutorial.md (hunk 7 above)
Comment (neutrinoks) [reply doc/tutorial.md]: Done

**Review by asmie — APPROVED** [review]
LGTM :+1:

**Pull request merged**